Sources/IO/Geometry/GLTFReader/ORMTexture.worker.js

```diff
@@ -0,0 +1,38 @@
+import registerWebworker from 'webworker-promise/lib/register';
+
+/**
+ *
+ * @param {ArrayBuffer} imageBuffer
+ * @param {string} mimeType
+ * @param {string} channel
+ * @returns {Promise<ImageData>}
+ */
+registerWebworker(async ({ imageBuffer, mimeType, channel }) => {
+  const t = Date.now();
+
+  const blob = new Blob([imageBuffer], { type: mimeType });
+  const imageUrl = URL.createObjectURL(blob);
+
+  const img = await createImageBitmap(blob);
+  const canvas = new OffscreenCanvas(img.width, img.height);
+  const ctx = canvas.getContext('2d');
+
+  ctx.drawImage(img, 0, 0, img.width, img.height);
+  const bitmap = ctx.getImageData(0, 0, img.width, img.height);
+
+  if (channel) {
+    let idx;
+    if (channel === 'r') idx = 0;
+    if (channel === 'g') idx = 1;
+    if (channel === 'b') idx = 2;
+
+    for (let i = 0; i < bitmap.data.length; i += 4) {
+      bitmap.data[i + idx] = bitmap.data[i + idx]; // Red channel
+      bitmap.data[i + ((idx + 1) % 3)] = bitmap.data[i + idx]; // Green channel
+      bitmap.data[i + ((idx + 2) % 3)] = bitmap.data[i + idx]; // Blue channel
```

Comment:
I'm not familiar with the action happening here, but I did want to comment on it just to double check the logic.

It looks like you are taking a particular channel of an RGBA pixel and making the pixel grayscale based on the selected channel value (e.g. if channel is "g", then you're turning pixels to be GGGA). If so, it's simpler to write it as follows (and the channel comments actually align with what channel you're writing to):

```
const channelValue = bitmap.data[i + idx]
bitmap.data[i] = channelValue // red channel
bitmap.data[i + 1] = channelValue // green channel
bitmap.data[i + 2] = channelValue // blue channel
```